### DISCUSSION TITLE
Problem by applying Config on Node behind Firewall

### DISCUSSION THREAD

**Discussion by 53845714nF** [siderolabs/talos#9047] [Q&A]
Hi, I have an Opensense Firewall before the Talos.

I try to apply the Config with this command:
```bash 
./talosctl apply-config --insecure --nodes $CONTROL_PLANE_IP --file _out/controlplane.yaml

```

I get this: 
```
error applying new configuration: rpc error: code = Unavailable desc = error reading from server: read tcp 192.168.178.181:53456->10.2.0.10:50000: read: connection timed out
```

I have the following Rule:
The allowed Ports are 50000 and 50001 als mentiond in: [Talos Network Conectivity](https://www.talos.dev/v1.7/learn-more/talos-network-connectivity/)
The WAN network is my network in which my computer is located.
![image](https://github.com/user-attachments/assets/7bec1b29-7bf0-438e-bf77-2dd33f3c287f)

The firewall blocks noting:
My IP is: 192.168.178.181
The Cluster IP is: 10.2.0.10

![image](https://github.com/user-attachments/assets/8b53beb1-e2a6-4f1e-b71b-9411918d3b28)

I looked in to Wireshark, there are a lot of TCP retransmissions. But I don't know why.

[wireshark.zip](https://github.com/user-attachments/files/16335814/wireshark.zip)

**Comment by smira** [comment]
Talos API runs over gRPC/TCP direct to the node on port 50000 in this case, so if something happens, it's within your network.

It's either the firewall, or something else, like wrong MTU. Talos doesn't block any connections in the default (empty) configuration.

You could use `talosctl disks --insecure --node $CONTROL_PLANE_IP` to simply test the connectivity without making changes.

**Comment by 53845714nF** [comment]
I get the right Output:
```
DEV        MODEL           SERIAL   TYPE   UUID   WWID   MODALIAS      NAME   SIZE    BUS_PATH                                                                   SUBSYSTEM          READ_ONLY   SYSTEM_DISK
/dev/sda   QEMU HARDDISK   -        HDD    -      -      scsi:t-0x00   -      22 GB   /pci0000:00/0000:00:05.0/0000:01:01.0/virtio3/host2/target2:0:0/2:0:0:0/   /sys/class/block 
``` 

But the Problem with the applying config is still there. 


(for the recording, it is `--nodes` not `--node`)

**Reply by smira** [reply]
my only guess is that you apply such machine config that the networking is reconfigured, and the response never comes back (e.g. because the IP is gone, or you enable firewall and break the connection).

But either way there's no problem with Talos here so far.

**Reply by 53845714nF** [reply]
Do I change the IP address of the node when I roll out a config? I had actually set this with DHCP.
If I interpret the Wireshark recording correctly, I don't see a problem with IP addresses, but rather that something is wrong with the TCP stream. 

 I also need to have another look at the network, the hint about the MTU could perhaps also play a role. The network is in a Proxmox SDN with Vxlan. The MTU is set automatically.

**Reply by smira** [reply]
you also have Talos console logs after the config is applied - if it reconfigures anything on the network side, it will be in the logs.

the problem might be with the MTU as well - big packets don't go through
